lib/scholar/metrics/regression.ex

```diff
@@ -519,6 +519,65 @@ defmodule Scholar.Metrics.Regression do
     Nx.reduce_max(Nx.abs(y_true - y_pred))
   end
 
+  @doc ~S"""
+  Calculates the mean pinball loss to evaluate predictive performance of quantile regression models.
+
+  $$pinball(y, \hat{y}) = \frac{1}{n) \sum_{i=1}^{n} \alpha max(\hat{y_i} - y_i, 0) +
+  (1 - \alpha) max(\hat{y_i} - y_i, 0)$$
+
+  The residual error is defined as $$|y - \hat{y}|$$ where $y$ is a true value
+  and $\hat{y}$ is a predicted value.
+  Equivalent to half of the $$mean_absolute_error$$ when $$\alpha$$ is 0.5.
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([1, 2, 3])
+      iex> y_pred = Nx.tensor([2, 3, 4])
+      iex> Scholar.Metrics.Regression.mean_pinball_loss(y_true, y_pred)
+      #Nx.Tensor<
+        f32
+        0.5
+      >
+      iex> y_true = Nx.tensor([[1, 0, 0, 1], [0, 1, 1, 1], [1, 1, 0, 1]])
+      iex> y_pred = Nx.tensor([[0, 0, 0, 1], [1, 0, 1, 1], [0, 0, 0, 1]])
+      iex> Scholar.Metrics.Regression.mean_pinball_loss(y_true, y_pred, alpha: 0.5, multioutput: :raw_values)
+      #Nx.Tensor<
+        f32[4]
+        [0.5, 0.3333333432674408, 0.0, 0.0]
+      >
+  """
+  defn mean_pinball_loss(y_true, y_pred, opts \\ [alpha: 0.5]) do
+    assert_same_shape!(y_true, y_pred)
+    alpha = opts[:alpha]
+
+    # Formula adapted from sklearn:
+    # https://github.com/scikit-learn/scikit-learn/blob/128e40ed593c57e8b9e57a4109928d58fa8bf359/sklearn/metrics/_regression.py#L299
+    diff = y_true - y_pred
+    sign = diff >= 0
+    loss = alpha * sign * diff - (1 - alpha) * (1 - sign) * diff
+
+    output_errors = handle_sample_weights(loss, opts, axes: [0])
+    # mimics the sklearn behavior
+    case opts[:multioutput] do
+      # raw_values returns plain output errors. One value per channel.
+      :raw_values -> output_errors
+      # uniform_average returns the mean of the above. Note how they are averaged.
+      :uniform_average ->
+        output_errors
+        |> Nx.mean()
+      # every other case, just take the mean without any axes nor weight.
+      _ -> handle_sample_weights(loss, opts)
```

Comment:
I think that for averaging errors we should have a separate tensor with weights, look: https://github.com/scikit-learn/scikit-learn/blob/128e40ed593c57e8b9e57a4109928d58fa8bf359/sklearn/metrics/_regression.py#L302-L309